src/wymeditor/editor/undo-redo.js

```diff
@@ -79,6 +79,7 @@ WYMeditor.UndoRedo.prototype._do = function (what) {
         if (history.changesetsBack.length === 0) {
             return;
         }
+        if (!history.changesetsFore.length) undoRedo._add()
```

Comment:
Also, the fact that we need to use an underscore method makes me think that either:

* This should be handled in `history` via a public method
* `_add()` should be `add()`

if the only existing implementation of a library requires usage of private methods, that's a good sign we can improve the API.